gitlabform/processors/project/files_processor.py

```diff
@@ -17,6 +17,25 @@
 
 
 class FilesProcessor(AbstractProcessor):
+    def pre_protection_check(self, project_and_group: str, configuration: dict):
+        all_branches = self.gitlab.get_branches(project_and_group)
+        sorted_branches = sorted(all_branches)
+        current_protected_state = sorted_branches["protected"]
+
+        desired_protected_state = True
+        desired_push_access_level = "no one"
+        desired_merge_access_level = "maintainers"
+
+        needs_unprotect = current_protected_state and (
+        sorted_branches["push_access_levels"][0]["access_level_description"] != desired_push_access_level or 
+        sorted_branches["merge_access_levels"][0]["access_level_description"] != desired_merge_access_level
+        )
+
+        if needs_unprotect:
```

Comment:
If branch is currently unprotected, then it should also never need unprotecting 🤔